compiler/circle-resizer/src/ModelEditor.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ModelEditor.h"
+
+#include <mio/circle/schema_generated.h>
+
+#include <loco/IR/Graph.h>
+#include <logo/Phase.h>
+#include <logo/RemoveDeadNodeWithQueryPass.h>
+#include <luci/IR/Nodes/CircleInput.h>
+#include <luci/Pass/CircleShapeInferencePass.h>
+#include <luci/Pass/CircleTypeInferencePass.h>
+
+using namespace circle_resizer;
+
+namespace
+{
+
+void change_single_input_shape(luci::CircleInput *circle_input, const Shape &new_shape)
```

Comment:
@jinevening One thing to confirm here. Should `circle-resizer` also modify `shape_signature` attribute? I think no because it's a something specific for TF but I would like to confirm.

Reply:
circle uses `shape_signature` as well. PTAL `luci/import/src/Importer.cpp`.

(Not 100% sure) But you may not have to do further modification, because shape signature may be automatically set by `encodeShapeSignature` in `CircleTensorExporter.cpp`.